apiserver/facades/client/secrets/register.go

```diff
@@ -63,6 +63,20 @@ func newSecretsAPI(context facade.Context) (*SecretsAPI, error) {
 		}
 		return p.NewBackend(cfg)
 	}
+
+	modelNameCache := make(map[string]string)
+	modelNameGetter := func(uuid string) (string, error) {
+		if name, ok := modelNameCache[uuid]; ok {
+			return name, nil
+		}
+		m, helper, err := context.StatePool().GetModel(uuid)
```

Comment:
We said we didn't want to do this in 4.0. We don't want you to peek into another model from a model. We should think about the correct way to do this. Obviously this is an established pattern in pre 4.0.


For 4.0 you probably can consult the controller db to get the model name.

Reply:
No longer use this